main.go

```diff
@@ -223,3 +322,26 @@ func post(ctx context.Context, endpoint *url.URL, token string, wreq *prompb.Wri
 	}
 	return nil
 }
+
+func successerrors() (float64, float64) {
+	metrics := make(chan prometheus.Metric, 2)
+	queryResponses.Collect(metrics)
+	close(metrics)
+
+	var success, errors float64
+
+	for m := range metrics {
+		m1 := &dto.Metric{}
+		_ = m.Write(m1)
```

Comment:
I don't want to surface this error, to be honest. We query the internal data structures here and the error that can occur is `fmt.Errorf("encountered unknown type %v", t)`. I don't think that this will happen without a compilation error in most cases anyway.

Reply:
Then the right thing to do here is log or panic, no?